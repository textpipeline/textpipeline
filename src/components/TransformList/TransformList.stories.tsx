import { storiesOf } from '@storybook/react';
import React from 'react';

import TransformList from './TransformList';

storiesOf('TransformList', module).add('default', () => <TransformList />);
