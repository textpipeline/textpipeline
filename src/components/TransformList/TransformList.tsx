import List from '@material-ui/core/List';
import EditIcon from '@material-ui/icons/Edit';
import React from 'react';

import { buildTransformPath } from '../../routes';
import transforms from '../../transforms';
import ListItemLink from '../ListItemLink';

const TransformList: React.FC = () => (
  <List>
    {Object.keys(transforms).map(slug => {
      const { name } = transforms[slug];
      return <ListItemLink key={slug} icon={EditIcon} text={name} href={buildTransformPath(slug)} />;
    })}
  </List>
);

export default TransformList;
