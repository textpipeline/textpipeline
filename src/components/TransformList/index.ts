export { default } from './TransformList';
export * from './TransformList';
