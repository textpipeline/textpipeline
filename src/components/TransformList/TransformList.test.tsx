import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import TransformList from './TransformList';

describe('TransformList', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<TransformList />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
