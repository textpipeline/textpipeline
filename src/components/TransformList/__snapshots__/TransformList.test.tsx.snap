// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformList renders correctly 1`] = `
<WithStyles(ForwardRef(List))>
  <ListItemLink
    href="/transform/csv-to-xls"
    icon={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "displayName": "EditIcon",
        "muiName": "SvgIcon",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    key="csv-to-xls"
    text="CSV to XLS"
  />
  <ListItemLink
    href="/transform/xls-to-csv"
    icon={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "displayName": "EditIcon",
        "muiName": "SvgIcon",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    key="xls-to-csv"
    text="XLS to CSV"
  />
</WithStyles(ForwardRef(List))>
`;
